src/xrUICore/XML/UIXmlInitBase.cpp

```diff
@@ -1058,32 +1058,62 @@ void CUIXmlInitBase::ApplyAlign(float& x, float& y, u32 align)
 
 //////////////////////////////////////////////////////////////////////////
 
+// source: https://stackoverflow.com/questions/650162/why-cant-the-switch-statement-be-applied-to-strings
+constexpr uint32_t hash(const std::string_view data) noexcept 
```

Comment:
@Xottab-DUTY can you advise where to put this function? It can be used to refactor other large if statement blocks with strings -> switch cases.

Reply:
Typically, this should be put somewhere in xrCore.